### PR TITLE
Added new tests for last timestamp filter

### PR DESCRIPTION
**Pull request by VVVyacheslav**

### CODE REVIEW

File: src/test/java/com/axibase/tsd/api/method/series/SeriesQueryLastTimestampFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+    @Issue("6112")
+    @Test (groups = "count", dataProvider = "all_dates_data", description = "Checks that response contain correct number of series")
+    public void testCountSeriesMinInsertDate(String minDate) {
+        SeriesQuery query = QUERY
+                .toBuilder()
+                .minInsertDate(minDate)
+                .build();
+        List<Series> seriesList = querySeriesAsList(query);
+        assertTrue(seriesList.size() <= 1, "Wrong count of series");
```
Comment (raipc) [inline src/test/java/com/axibase/tsd/api/method/series/SeriesQueryLastTimestampFilterTest.java]: Assertions must be more clear: check if size is either 0 or either 1. Expected size may be provided via parameter

File: src/test/java/com/axibase/tsd/api/method/series/SeriesQueryLastTimestampFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    public void testLastTimestampEqualOrGreaterMaxInsertDate(String data) {
+        SeriesQuery query = QUERY
+                .toBuilder()
+                .maxInsertDate(data)
+                .build();
+        List<Series> seriesList = querySeriesAsList(query);
+        assertTrue(seriesList.isEmpty() || seriesList.get(0).getData().isEmpty(), "Output series not empty");
+    }
+
+    private void addSamplesToSeries(Series series) {
```
[the remark this answers is not in the document]
Comment (Arthasasyan) [reply src/test/java/com/axibase/tsd/api/method/series/SeriesQueryLastTimestampFilterTest.java]: Inlined

File: src/test/java/com/axibase/tsd/api/method/series/SeriesQueryLastTimestampFilterTest.java (hunk 1 above)
Comment (Arthasasyan) [reply src/test/java/com/axibase/tsd/api/method/series/SeriesQueryLastTimestampFilterTest.java]: Please, check if current assertions are ok

**Comment by Arthasasyan** [comment]
> delete all test methods of the "count" group,

`count` group tests could be either inlined in corresponding methods or deleted, because assertions in these corresponding methods will fail in any case if assertions from `count` group fail. For example, if this assertion from `count` group method fail:

```java
assertTrue(seriesList.size() == 0 || seriesList.size() == 1, "Wrong count of series");
```
assertion from corresponding method will fail too:

```java
assertTrue(seriesList.isEmpty() || seriesList.get(0).getData().isEmpty(), "Output series not empty");
```

@raipc @MikhailZvagelsky what do you think should be done with them?

**Comment by raipc** [comment]
I added a note about multiple assertions in the [styleguide](https://github.com/axibase/atsd-api-test/blob/styleguide/STYLEGUIDE.md#66-multiple-assertions-in-tests-allowed)

File: src/test/java/com/axibase/tsd/api/method/series/SeriesQueryLastTimestampFilterTest.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    @Issue("6112")
+    @Test (dependsOnGroups = "count", dataProvider = "date_less_or_equals_last_data",
+            description = "Checks that output series is empty if last series timestamp equal or greater than maxInsertDate")
+    public void testLastTimestampEqualOrGreaterMaxInsertDate(String maxDate) {
+        SeriesQuery query = QUERY.withMaxInsertDate(maxDate);
+        List<Series> seriesList = querySeriesAsList(query);
+        assertTrue(seriesList.isEmpty() || seriesList.get(0).getData().isEmpty(), "Output series not empty");
+    }
+
+    private void setDatePairs() {
```
Comment (raipc) [inline src/test/java/com/axibase/tsd/api/method/series/SeriesQueryLastTimestampFilterTest.java]: Please, follow the [styleguide](https://github.com/axibase/atsd-api-test/blob/styleguide/STYLEGUIDE.md#65-static-members-never-modify-after-initialization)

Comment (Arthasasyan) [reply src/test/java/com/axibase/tsd/api/method/series/SeriesQueryLastTimestampFilterTest.java]: I suggest changing it to this:

```java
private static final List<String[]> DATE_PAIRS = new ArrayList<>();

    static {
        for (String minDate: DATES) {
            for (String maxDate: DATES) {
                DATE_PAIRS.add(new String[] {minDate, maxDate});
            }
        }
    }
```

Comment (unrealwork) [reply src/test/java/com/axibase/tsd/api/method/series/SeriesQueryLastTimestampFilterTest.java]: @Arthasasyan 

```java
private static final List<String[]> DATE_PAIRS = Arrays.stream(DATES)
            .flatMap(d1 -> Arrays.stream(DATES).map(d2 -> toArray(d1, d2)))
            .collect(toList());
```

Comment (raipc) [reply src/test/java/com/axibase/tsd/api/method/series/SeriesQueryLastTimestampFilterTest.java]: As for me, moving it to a method `List<String[]> createDatePermutations(String[] dates)` seems to be a more correct option

**Pull request merged**